ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/AccountTrieNodeDataRequest.java

```diff
@@ -43,7 +43,12 @@ protected void doPersist(final Updater updater) {
 
   @Override
   public Optional<Bytes> getExistingData(final WorldStateStorage worldStateStorage) {
-    return worldStateStorage.getAccountTrieNodeData(getLocation().orElse(Bytes.EMPTY), getHash());
+    return getLocation()
+        .flatMap(
+            location ->
+                worldStateStorage
+                    .getAccountStateTrieNode(location, getHash())
+                    .filter(data -> Hash.hash(data).equals(getHash())));
```

Comment:
if I understood well, when the data at that location does not match the hash, that location could be deleted?

Reply:
it could, and we could implement lazy writes also.  For now though we simply rely on subsequent writes to clean up a dirty value.